### PR TITLE
HW complete

### PR DESCRIPTION
**Pull request by thewildjacko**

### CODE REVIEW

File: week7/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+g) How can you find the number of bytes a Data Type uses?</br>
+**with `MemoryLayout.size(ofValue:)`, or for arrays, by multiplying the `MemoryLayout.size` by the array's `count`**
+
+h) Using Playgrounds, how can you tell that the Data.write operation succeeded?</br>
+**The result will appear as a readout in the sidebar, and a file will appear in the document directory.**
+
+i) You can mostly treat Data objects like **arrays** of bytes.
+
+j) The write and read methods of Data require a **try**.
```
Comment (byaruhaf) [inline week7/README.md:35]: URL

File: week7/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+k) What JavaScript calls an object is the same concept as a heterogenous **dictionary** in Swift with **strings** for keys.
+
+l) How do you resolve the error: Use of unresolved identifier ‘Bundle’?</br>
+**`import Foundation`**
+
+m) Give an example of Snake Case. **json_is_difficult_but_i_kind_of_understand_it_now**
+
+n) A struct that will be used in the reading and writing of data must conform to the **Codable** Protocol
+
+o) Show the line of code used to access the user’s document directory for the running app. **`print(FileManager.documentsDirectoryURL)`**
```
Comment (byaruhaf) [inline week7/README.md:46]:
```swift
let documentDirectory = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!

In iOS 16 and above, URL.documentsDirectory can be used instead.
```

File: week7/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+p) Files added to the project that will be used by the app can be found in the **app bundle** when the app is running.
```
Comment (byaruhaf) [inline week7/README.md:48]: Main bundle

File: week7/Homework7-start/Homework7/Models/API/APIStore.swift [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  @Published var names: APIS = APIS(apis: []) {
+    didSet {
+      saveJSON()
+    }
+  }
+  
+  let apisJSONURL = URL(fileURLWithPath: "apilist", relativeTo: FileManager.documentsDirectoryURL)
+    .appendingPathExtension("json")
+  
+  init() {
```
Comment (byaruhaf) [inline week7/Homework7-start/Homework7/Models/API/APIStore.swift:57]: I noticed that `loadJSON()` isn't being called in the `onAppear` modifier within your SwiftUI view. Calling `loadJSON()` in `onAppear` ensures that your data is loaded when the view appears. Was there a specific reason for not doing this? 

Here's an example of how you could incorporate `loadJSON()` into the `onAppear` modifier:

```swift
struct ContentView: View {
    @State private var data: [YourDataType] = []

    var body: some View {
        List(data) { item in
            Text(item.name) // Customize this line based on your data structure
        }
        .onAppear {
            loadJSON()
        }
    }
```

This way, `loadJSON()` will be called when the view appears, ensuring that your data is loaded appropriately.

File: week7/Homework7-start/Homework7/Models/User/UserStore.swift [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  ) {
+    didSet {
+      saveJSON()
+    }
+  }
+  
+  let userJSONURL = URL(fileURLWithPath: "1-week7aboveandbeyond", relativeTo: FileManager.documentsDirectoryURL)
+    .appendingPathExtension("json")
+  
+  init() {
```
Comment (byaruhaf) [inline week7/Homework7-start/Homework7/Models/User/UserStore.swift:54]: See last comment on loadJason

**Review by byaruhaf — APPROVED** [review]
Great job on your work, Jake! only have one comment for improvement;  Keep up the excellent work!

File: week7/README.md (hunk 2 above)
Comment (thewildjacko) [reply week7/README.md:46]: oh ok good to know, thanks!

File: week7/Homework7-start/Homework7/Models/API/APIStore.swift (hunk 4 above)
Comment (thewildjacko) [reply week7/Homework7-start/Homework7/Models/API/APIStore.swift:57]: Thanks, this is helpful!

**Pull request merged**